clcache.py

```diff
@@ -206,7 +206,7 @@ def computeKey(compilerBinary, commandLine):
         (preprocessedSourceCode, pperr) = preprocessor.communicate()
 
         if preprocessor.returncode != 0:
-            sys.stderr.write(pperr)
+            sys.stderr.write(pperr.decode("utf-8"))
```

Comment:
Promlem is in sys.stderr  API.
sys.stderr.write(perr) works for python 2.7 compatible. 
sys.stderr.writebuffer(perr) for python 3 (or somthing like it).
I just do bit know how to forward bytes to stderr for both versions without explicity decoding.